dask/array/tests/test_atop.py

```diff
@@ -202,6 +202,26 @@ def test_optimize_blockwise():
     )
 
 
+def test_optimize_blockwise_annotations():
+    x = da.ones(10, chunks=(5,))
+    u = x + 1
+    v = u + 2
+
+    with dask.annotate(qux="bax"):
+        w = v + 3
+
+    y = w + 4
+    z = y + 5
+
+    dsk = da.optimization.optimize_blockwise(z.dask)
+
+    assert isinstance(dsk, HighLevelGraph)
+    assert (
+        len([layer for layer in dsk.dicts.values() if isinstance(layer, Blockwise)])
+        == 3
+    )
```

Comment:
I don't understand why this is 3. Can you help explain?

I *think* that maybe you're trying to test that layers with annotations don't fuse with other layers with mismatching annotations.  If so, I wonder if maybe we could have a test with two layers with different annotations, and then assert that exactly one layer with each annotation still exists in the resulting high level graph.

Reply:
Changed to assert the counts of the final number of unique annotations